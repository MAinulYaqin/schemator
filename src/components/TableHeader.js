import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import "typeface-roboto";

const Header = styled.div`
  width: 100%;
  height: 32px;
  margin-bottom: 4px;
  background: #b33939;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
`;

const Caption = styled.p`
  color: white;
  font-size: 12px;
  font-family: Roboto;
  font-weight: bold;
  user-select: none;
`;

const TableHeader = ({ caption }) => (
  <Header>
    <Caption>{caption}</Caption>
  </Header>
);

TableHeader.propTypes = {
  caption: PropTypes.string
};

export default TableHeader;
