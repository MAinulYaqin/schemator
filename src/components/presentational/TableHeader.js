import React, { Component } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import "typeface-roboto";

const Header = styled.div`
  width: 100%;
  height: 32px;
  margin-bottom: 4px;
  background: #b33939;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
`;

export const Caption = styled.input`
  color: white;
  border: 0px;
  width: 50%;
  outline: none;
  background: none;
  font-size: 12px;
  font-family: Roboto;
  font-weight: bold;
  text-align: center;
`;

class TableHeader extends Component {
  shouldComponentUpdate(nextProps) {
    const { caption } = this.props;

    return nextProps.caption !== caption;
  }

  render() {
    const { caption, onChangeCaption } = this.props;

    return (
      <Header>
        <Caption type="text" value={caption} onChange={onChangeCaption} />
      </Header>
    );
  }
}

TableHeader.propTypes = {
  caption: PropTypes.string,
  onChangeCaption: PropTypes.func
};

export default TableHeader;
