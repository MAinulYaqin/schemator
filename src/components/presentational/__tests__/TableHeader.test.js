import React from "react";
import { shallow } from "enzyme";

import TableHeader, { Caption } from "../TableHeader";

describe("TableHeader", () => {
  const setup = propOverrides => {
    const props = {
      caption: "User",
      onChangeCaption: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<TableHeader {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should calls 'onChangeCaption' while caption changed", () => {
    const { wrapper, props } = setup();
    const event = {
      target: {
        value: "Comment"
      }
    };

    wrapper.find(Caption).simulate("change", event);

    expect(props.onChangeCaption).toBeCalledWith(event);
  });
});
