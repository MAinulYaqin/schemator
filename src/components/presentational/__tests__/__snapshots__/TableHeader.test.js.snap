// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader should renders default 1`] = `
<styled.div>
  <styled.input
    onChange={[Function]}
    type="text"
    value="User"
  />
</styled.div>
`;
